### PR TITLE
Support for Entity.show

### PR DESCRIPTION
**Pull request by mramato**
This change is way less scary than it may look at first, it's just that adding a top-level entity visibility property ended up with small changes throughout the code.

1. Added `Entity.show` which is a boolean for easily hiding or showing an entity and its children.
2. Added `Entity.isShowing` which is a read-only property that indicates if an entity is currently being drawn (i.e. its show property and all ancestor show properties are true).  This is mainly used by the visualizers but could be useful for other stuff as well.
3. Added support for the KML `visibility` element, which is mapped directly to `Entity.show`.
4. Added a new beginner Sandcastle example for toggling entity visibility.
5. To support the Sandcastle example, I changed `viewer.zoomTo` and `viewer.flyTo` to be "best effort" functions, rather than failing if any one entity is not displayed. This was both requested on the mailing and really an oversight on my part when I initial implemented them.  For example, before this change a KML Folder would create an entity with no visualization but with children, zooming to the data would fail as soon as it found the folder, now it ignore the folder and zooms to all data that is actually visualized.  This was a one line code change, so don't be alarmed by the length of this bullet point.
6. On an architectural note, Entities, which previously only knew about their parent, now also know about their children. This was needed to properly propagate setting a parent's show property without dealing with tons of notification subscriptions. I didn't expose this publicly yet because I want to make sure there aren't ramifications that I didn't think of, but I wanted to mention it here in case anyone had any thoughts on the matter.

### CODE REVIEW

**Comment by mramato** [comment]
Issue 2600?  Where's my Cap'n Crunch whistle?

**Comment by pjcozzi** [comment]
All looks good.

File: Source/DataSources/Entity.js [hunk 1]
```diff
@@ -176,10 +177,26 @@ define([
...
+        for (var i = 0; i < length; i++) {
+            var child = children[i];
+            var childShow = child._show;
+            var oldValue = !isShowing && childShow;
+            var newValue = isShowing && childShow;
+            if (oldValue !== newValue) {
+                child._definitionChanged.raiseEvent(child, 'isShowing', newValue, oldValue);
+            }
+        }
+        entity._definitionChanged.raiseEvent(entity, 'isShowing', isShowing, !isShowing);
```
[the remark this answers is not in the document]
Comment (sceptre12) [reply Source/DataSources/Entity.js]: The issue occurs when dealing with entities from the CompositeEntityCollection. Its own CompositeEntityCollection.prototype._onDefinitionChanged throws an error whenever " entity._definitionChanged.raiseEvent(entity, 'isShowing', isShowing, !isShowing)" event is raised

[the remark this answers is not in the document]
Comment (OmarShehata) [reply Source/DataSources/Entity.js]: @sceptre12  if you can reproduce this issue/error in a [Sandcastle example](https://cesiumjs.org/Cesium/Apps/Sandcastle/index.html), please open a new GitHub issue for it.

**Pull request merged**